arduino/OpenMRNLite.h

```diff
@@ -343,36 +344,69 @@ class OpenMRN : private Executable
     {
         for (auto *e : loopMembers_)
         {
-#if defined(ESP32)
+#if defined(ESP32) && CONFIG_TASK_WDT
             // Feed the watchdog so it doesn't reset the ESP32
             esp_task_wdt_reset();
-#endif // ESP32
+#endif // ESP32 && CONFIG_TASK_WDT
             e->run();
         }
     }
 
 #ifndef OPENMRN_FEATURE_SINGLE_THREADED
+    /// Entry point for the executor thread when @ref start_executor_thread is
+    /// called with donate_current_thread set to false.
     static void thread_entry(void *arg)
     {
         OpenMRN *p = (OpenMRN *)arg;
         p->stack()->executor()->thread_body();
     }
 
+    /// Donates the calling thread to the @ref Executor.
+    ///
+    /// Note: this method will not return until the @ref Executor has shutdown.
+    void loop_executor()
+    {
+#if defined(ESP32) && CONFIG_TASK_WDT
+        uint32_t current_core = xPortGetCoreID();
+        TaskHandle_t idleTask = xTaskGetIdleTaskHandleForCPU(current_core);
+        // check if watchdog is enabled and print a warning if it is
+        if (esp_task_wdt_status(idleTask) == ESP_OK)
+        {
+            LOG(WARNING, "WDT detected as enabled on core %d!", current_core);
+        }
+#endif // ESP32 && CONFIG_TASK_WDT
+        haveExecutorThread_ = true;
+
+        // donate this thread to the executor
+        stack_->executor()->thread_body();
```

Comment:
pleae do stack_->loop_executor() instead of calling this directly.

Reply:
that would call start_stack() again, is that the intent here? In the thread_entry method it also calls like this.

Reply:
also should thread_entry call this method?

Reply:
I tested locally and calling start_stack again has no impact so will switch accordingly. Open q on thread_entry calling this method instead of duplication of code.

Reply:
apologies this was a bad suggestion from me. calling start_stack(...) more than once is not a good idea. I forgot that we already have a begin() call which does that. Since we do not want to retire support for begin() + loop() we should have as you originally write the code.